api/provider/aws/job_runner.go

```diff
@@ -152,6 +156,32 @@ func (r *JobRunner) createTask(jobID, request string) (string, error) {
 	})
 }
 
+func (r *JobRunner) linkEnvironment(jobID, request string) (string, error) {
+	var req models.CreateEnvironmentLinkRequest
+	if err := json.Unmarshal([]byte(request), &req); err != nil {
+		return "", errors.New(errors.InvalidRequest, err)
+	}
+
+	if err := r.environmentProvider.Link(req); err != nil {
+		return "", err
+	}
+
+	return req.SourceEnvironmentID, nil
+}
+
+func (r *JobRunner) unlinkEnvironment(jobID, request string) (string, error) {
+	var req models.DeleteEnvironmentLinkRequest
+	if err := json.Unmarshal([]byte(request), &req); err != nil {
+		return "", errors.New(errors.InvalidRequest, err)
+	}
+
+	if err := r.environmentProvider.Unlink(req); err != nil {
+		return "", err
+	}
+
+	return req.SourceEnvironmentID, nil
```

Comment:
Any idea for better names than `Source` and `Dest`? The implied direction problem you brought up should probably be addressed in this pr.